crates/sui-protocol-constants/src/lib.rs

```diff
@@ -0,0 +1,87 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+// Constants that change the behavior of the protocol
+
+// ==== Move VM, Move bytecode verifier, and execution limits ===
+
+/// Maximum Move bytecode version the VM understands. All older versions are accepted.
+pub const MOVE_BINARY_FORMAT_VERSION: u32 = 6;
+
+/// Maximum size of the `contents` part of an object, in bytes. Enforced by the Sui adapter when effects are produced.
+pub const MAX_MOVE_OBJECT_SIZE: u64 = 250 * 1024; // 250 KB
+
+// TODO: increase to 500 KB. currently, publishing a package > 500 KB exceeds the max computation gas cost
+/// Maximum size of a Move package object, in bytes. Enforced by the Sui adapter at the end of a publish transaction.
+pub const MAX_MOVE_PACKAGE_SIZE: u64 = 100 * 1024; // 100 KB
+
+/// Maximum number of gas units that a single MoveCall transaction can use. Enforced by the Sui adapter.
+pub const MAX_TX_GAS: u64 = 1_000_000_000;
+
+/// Maximum number of nested loops. Enforced by the Move bytecode verifier.
+pub const MAX_LOOP_DEPTH: usize = 5;
+
+/// Maximum number of type arguments that can be bound to generic type parameters. Enforced by the Move bytecode verifier.
+pub const MAX_GENERIC_INSTANTIATION_LENGTH: usize = 32;
+
+/// Maximum number of paramters that a Move function can have. Enforced by the Move bytecode verifier.
+pub const MAX_FUNCTION_PARAMETERS: usize = 128;
+
+/// Maximum number of basic blocks that a Move function can have. Enforced by the Move bytecode verifier.
+pub const MAX_BASIC_BLOCKS: usize = 1024;
+
+/// Maximum number of events that a single Move function can emit. Enforced by the Sui adapter during execution.
+// TODO: is this per Move function, or per transaction? And if per-function, can't I get around the limit by calling
```

Comment:
@oxade this question is for you.

I am also wondering where we enforce this (if we do enforce it). The only place I see this constant used is in the tables--the adapter never reads it:
```
λ git grep MAX_NUM_EVENT_EMIT
crates/sui-cost-tables/src/natives_tables.rs:            GasCost::new(MAX_TX_GAS / MAX_NUM_EVENT_EMIT, 1),
crates/sui-protocol-constants/src/lib.rs:pub const MAX_NUM_EVENT_EMIT: u64 = 256;
```

Reply:
I see this related task, which is marked as completed: https://github.com/MystenLabs/sui/issues/3341